File: Napakalaki RUBY/lib/numeric_bad_consequence.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame
class NumericBadConsequence < BadConsequence
  def initialize(t, l, nv, nh)
    super(t, l)
    @n_visible_treasures = nv
    @n_hidden_treasures = nh
  end
  
  def getVisibleTreasures()
    @n_visible_treasures
  end
  
  def getHiddenTreasures()
    @n_hidden_treasures
  end
  
  def isEmpty()
    if(@n_visible_treasures == 0 && @n_hidden_treasures == 0)
      return true
    end
    
    return false
  end
  
  def substractVisibleTreasure(t)
    if(@n_visible_treasures != 0)
        @n_visible_treasures = @n_visible_treasures - 1;
    end
  end
  
  def substractHiddenTreasure(t)
    if(@n_hidden_treasures != 0)
        @n_hidden_treasures = @n_hidden_treasures - 1;
    end
  end
  
  def adjustToFitTreasureLists(v, h)
    if(@n_visible_treasures <= v.size)
        nuevoNVisible = @n_visible_treasures
    else
        nuevoNVisible = v.size
    end
      
    if(@n_hidden_treasures <= h.size)
        nuevoNHidden = @n_hidden_treasures
    else
        nuevoNHidden = h.size
    end
    
    bc = NumericBadConsequence.new(getText(), getLevels(), nuevoNVisible, nuevoNHidden)
    
    return bc
  end
  
  def to_s
    return "\n " + getText() + "\n Pierdes " + getLevels().to_s + " niveles\n Tesoros visibles perdidos: #{@n_visible_treasures}\n Tesoros escondidos perdidos: #{@n_hidden_treasures}\n"
  end
end
end

File: Napakalaki JAVA/src/napakalaki_java/NumericBadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki_java;

import java.util.ArrayList;

/**
 *
 * @author alvaro89
 */
public class NumericBadConsequence extends BadConsequence {
    protected int nVisibleTreasures;
    protected int nHiddenTreasures;
    
    public NumericBadConsequence(String text, int levels, int nVisible, int nHidden){
        super(text,levels);
        nVisibleTreasures = nVisible;
        nHiddenTreasures = nHidden;
    }
    
    @Override
    public boolean isEmpty(){
        return (nVisibleTreasures == 0 && nHiddenTreasures == 0);
    }
    
    public int getVisibleTreasures(){
        return nVisibleTreasures;
    }
    
    public int getHiddenTreasures(){
        return nHiddenTreasures;
    }
    
    @Override
    public void substractVisibleTreasure(Treasure t){
        if(nVisibleTreasures != 0)
            nVisibleTreasures--; 
    }
    
    @Override
    public void substractHiddenTreasure(Treasure t){
        if(nHiddenTreasures != 0)
            nHiddenTreasures--;
    }
    
    @Override    
    public BadConsequence adjustToFitTreasureLists(ArrayList<Treasure> v, ArrayList<Treasure> h){
        BadConsequence nuevaBadConsequence;
        int nuevoNVisible, nuevoNHidden;

        if(nVisibleTreasures <= v.size())
            nuevoNVisible = nVisibleTreasures;
        else
            nuevoNVisible = v.size();

        if(nHiddenTreasures <= h.size())
            nuevoNHidden = nHiddenTreasures;
        else
            nuevoNHidden = h.size();

        nuevaBadConsequence = new NumericBadConsequence(text,levels,nuevoNVisible, nuevoNHidden);

        
        return nuevaBadConsequence;
    }
    
    @Override
    public String toString(){
    
        return text + " \nNiveles que pierdes = " + Integer.toString(levels) +
               "\nNumero visible treasures = " + Integer.toString(nVisibleTreasures) +
               "\nNumero hidden treasures = " + Integer.toString(nHiddenTreasures);
    }
    
}
